src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -581,10 +583,73 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
     auto config = selectedPd->getConfig();
     config.outConfs.front().setMemDesc(mem_desc);
-    //bypass any checks, we enforce the child descriptor
+    // bypass any checks, we enforce the child descriptor
     selectedPd->setConfig(config);
 }
 
+void MemoryInput::selectOptimalPrimitiveDescriptor() {
+    MemoryInputBase::selectOptimalPrimitiveDescriptor();
+    if (haveSubgraph()) {
+        std::vector<Input::InputConfig> graphInputConfig;
+        std::vector<PortConfig> inConfs;
+
+        for (size_t i = 0; i < getParentEdges().size(); i++) {
+            // MemoryInputBase::selectOptimalPrimitiveDescriptor() can't deduce corrent primitave.
+            auto desc = getParentOutputMemDesc(getParentEdgeAt(i));
+            inConfs.emplace_back(desc);
+            graphInputConfig.emplace_back(node::Input::InputConfig{desc, true});
+        }
+
+        std::vector<Input::OutputConfig> graphOutputConfig;
+        for (size_t i = 0; i < getChildEdges().size(); i++) {
+            // TODO: Upgrade OutputConfig and update OutputConfig's first param like InputConfig
+            graphOutputConfig.emplace_back(node::Input::OutputConfig{false, false});
+        }
+
+        // configure the inner graph to get the information about output memory descriptors
+        subGraph.Init(body, context, graphInputConfig, graphOutputConfig);
+
+        // for the output decriptors, use the configuration of the graph's output nodes
+        auto outputDescriptors = subGraph.getOutputMemoryDescriptors();
+
+        std::vector<PortConfig> outConfs;
+        for (const auto& desc : outputDescriptors) {
+            outConfs.emplace_back(desc);
+        }
+
+        const NodeConfig config(inConfs, outConfs);
+
+        supportedPrimitiveDescriptors.clear();
+        supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::undef);
+
+        selectPrimitiveDescriptorByIndex(0);
+    }
+}
+
+// @todo add ascii diagramm for memory mapping / reuse
+void MemoryInput::createPrimitive() {
+    MemoryInputBase::createPrimitive();
+    if (haveSubgraph()) {
+        OPENVINO_ASSERT(getOriginalInputsNumber() == subGraph.GetInputNodesMap().size(),
+                        "Number of node inputs must be equal the number of inner graph's inputs");
+
+        std::vector<MemoryPtr> inputMemory;
+        for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
+            inputMemory.emplace_back(getSrcMemoryAtPort(i));
+        }
+
+        OPENVINO_ASSERT(getOriginalOutputsNumber() == subGraph.GetOutputNodesMap().size(),
+                        "Number of node outputs must be equal the number of inner graph's outputs");
+
+        std::vector<MemoryPtr> outputMemory;
+        for (size_t i = 0; i < getOriginalOutputsNumber(); i++) {
+            outputMemory.emplace_back(getDstMemoryAtPort(i));
+        }
```

Comment:
As already said, this action is dangerous as the memory layout may differ between the output edge and the inner graph output. To prove this, simply replace the MatMul operation with Convolution.

Reply:
Add a new test based on Convolution. And also add shape inference for `MemoryInputSingle`.(old fallback shape inference maybe deduce out error shape).